meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -255,6 +256,176 @@ IMAGE_CMD_resinos-img () {
+		dst="$(echo ${RESIN_BOOT_PARTITION_FILE} | awk -F: '{print $2}')"
+		if [ -z "${dst}" ]; then
+			dst="/${src}" # dst was omitted
+		fi
+		src="${DEPLOY_DIR_IMAGE}/$src" # src is relative to deploy dir
+
+		# Check that dst is an absolute path and assess if it should be a directory
+		case $dst in
+			/*)
+				# Check if dst is a directory. Directory path ends with '/'.
+				case $dst in
+					*/) dst_is_dir=true ;;
+					*) dst_is_dir=false ;;
+				esac
+				;;
+			*) bbfatal "$dst in RESIN_BOOT_PARTITION_FILES is not an absolute path."
+		esac
+
+		# Check src type and existence
+		if [ -d "$src" ]; then
+			if ! $dst_is_dir; then
+				bbfatal "You can't copy a directory to a file. You requested to copy $src in $dst."
+			fi
+			sources="$(find $src -maxdepth 1 -type f)"
+		elif [ -f "$src" ]; then
+			sources="$src"
+		else
+			bbfatal "$src is an invalid path referenced in RESIN_BOOT_PARTITION_FILES."
+		fi
+
+		# Normalize paths
+		dst=$(realpath -ms $dst)
+		if $dst_is_dir && [ ! "$dst" = "/" ]; then
+			dst="$dst/" # realpath removes last '/' which we need to instruct mcopy that destination is a directory
+		fi
+		src=$(realpath -m $src)
+
+		for src in $sources; do
+			echo "Copying $src -> $dst ..."
+			# Create the directories parent directories in dst
+			directory="${RESIN_BOOT_DIR}"
+			for path_segment in $(echo ${dst} | sed 's|/|\n|g' | head -n -1); do
+				if [ -z "$path_segment" ]; then
+					continue
+				fi
+				directory=$directory/$path_segment
+				mkdir $directory || true
+			done
+			cp -r ${src} ${RESIN_BOOT_DIR}/${dst}
+			if $dst_is_dir; then
+				md5sum ${src} | awk -v awkdst="$dst$(basename $src)" '{ $2=awkdst; print }' >> ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+			else
+				md5sum ${src} | awk -v awkdst="$dst" '{ $2=awkdst; print }' >> ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+			fi
+		done
+	done
+	# Add stamp file
+	echo "${IMAGE_NAME}-${IMAGEDATESTAMP}" > ${WORKDIR}/image-version-info
+	cp ${WORKDIR}/image-version-info ${RESIN_BOOT_DIR}
+	md5sum ${WORKDIR}/image-version-info | awk -v filepath="/image-version-info" '{ $2=filepath; print }' >> ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+	# Finally add the fingerprint to boot
+	cp ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT} ${RESIN_BOOT_DIR}
+
+	init_config_json ${WORKDIR}
+
+	cp ${WORKDIR}/config.json ${RESIN_BOOT_DIR}/config.json.init
+
+	# Split MKUBIFS_ARGS_<name>
+	for name in ${UBIMULTIVOL_BUILD}; do
+		eval local mkubifs_args=\"\$MKUBIFS_ARGS_${name}\"
+		eval local additional_mkubifs_args=\"\$ADDITIONAL_MKUBIFS_ARGS_${name}\"
+
+		mkfs.ubifs ${additional_mkubifs_args} ${mkubifs_args}
+
+	done
+
+	RESIN_ROOT_SIZE=$(du -bks ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfsA.ubifs | awk '{print $1}')
+	ADDITIONAL_UBINIZE_ARGS_rootfsA="mode=ubi\nimage=${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfsA.ubifs\nvol_id=1\nvol_size=${RESIN_ROOT_SIZE}KiB\nvol_type=dynamic\nvol_name=${RESIN_ROOTA_FS_LABEL}"
```

Comment:
RESIN_ROOT_SIZE doesn't seem to be aligned either. I'm not sure if that hits performance but I'd say it would at least avoid discrepancies with the block images.